plugins/woocommerce-admin/client/marketplace/components/notice/notice.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { useState } from '@wordpress/element';
+import { Icon, check, closeSmall, info, percent } from '@wordpress/icons'; // See: https://wordpress.github.io/gutenberg/?path=/docs/icons-icon--docs
+
+/**
+ * Internal dependencies
+ */
+import './notice.scss';
+import sanitizeHTML from '../../../lib/sanitize-html';
+
+export interface NoticeProps {
+	id: string;
+	children?: JSX.Element;
+	description: string;
+	icon?: string;
+	isDismissible: boolean;
+	variant: string;
+}
+
+type IconKey = keyof typeof iconMap;
+
+// Map the icon name (string) to the actual icon component
+const iconMap = {
+	info,
+	check,
+	percent,
+};
+
+export default function Notice( props: NoticeProps ): JSX.Element {
+	const {
+		id,
+		description,
+		children,
+		icon,
+		isDismissible = true,
+		variant = 'info',
+	} = props;
+	const [ isVisible, setIsVisible ] = useState(
+		localStorage.getItem( `wc-marketplaceNoticeClosed-${ id }` ) !== 'true'
+	);
+
+	const handleClose = () => {
+		setIsVisible( false );
+		localStorage.setItem( `wc-marketplaceNoticeClosed-${ id }`, 'true' );
+	};
+
+	if ( ! isVisible ) return <></>;
+
+	const classes = classNames( 'woocommerce-marketplace__notice', {
+		'woocommerce-marketplace__notice-warning': variant === 'warning',
+		'woocommerce-marketplace__notice-info':
+			variant === 'info' || variant === 'info' || variant === 'error',
+		'woocommerce-marketplace__notice-error': variant === 'error',
+		'woocommerce-marketplace__notice-success': variant === 'success',
+	} );
+
+	const iconElement = iconMap[ ( icon || 'info' ) as IconKey ];
+
+	// Add a class for the icon color based on the variant for consistency.
+	const iconClass = classNames( 'woocommerce-marketplace__notice-icon', {
+		'icon-color-info': variant === 'info',
+		'icon-color-error': variant === 'error',
+		'icon-color-warning': variant === 'warning',
+		'icon-color-success': variant === 'success',
```

Comment:
For consistency with the parent component class name I would suggest using the following string for a class name `woocommerce-marketplace__notice-icon--${ variant }`. Also, both here and in line 52, parts related to variant (info, error, warning, success) look like element [modifiers](https://getbem.com/naming/#modifier) and I see that it's a convention used in WooCommerce Core codebase, so maybe we should use it here too?

Reply:
It also occurred to me we could do these as modifiers.